### ISSUE TITLE
[Tutorial] Autocomplete after delete input

### ISSUE THREAD

**Issue by ChangJoo-Park** [emberjs/guides#1236]
I am following ember.js 2.3.0 guide.

now I read and code autocomplete method.
https://guides.emberjs.com/v2.3.0/tutorial/autocomplete-component/

In this page, **app/controller/index.js** make undefined error when empty input string.

I don't want to leave it.

So.

is it right code to solve?
I find Ember.isEmpty().

```app/controllers/index.js
export default Ember.Controller.extend({
  filteredList: null,
  actions: {
    autoComplete(param) {
      if(param !== "") {
        this.store.query('rental', {city: param}).then((result) => {
          this.set('filteredList',result);
        });
      }
      else {
        let filteredList = this.get('filteredList');
        if(Ember.isEmpty(filteredList)) {
          return;
        }
        console.log();
        this.set('filteredList').clear();
      }
    },
    search(param) {
      if(param !== "") {
        this.store.query('rental', {city: param}).then((result) => {
          this.set('model',result);
        });
      }
      else {
        let model = this.get('model');
        if(Ember.isEmpty(model)){
          return;
        }
        this.set('model').clear();
      }
    }
  }
});

```
I want to know someone who have elegant solution.

**add 2016-02-09**
I solved problem but I don't trust my code is ember.js way.

```javascript
export default Ember.Controller.extend({
  filteredList: null,
  actions: {
    autoComplete(param) {
      if(param !== "") {
        this.store.query('rental', {city: param}).then((result) => {
          this.set('filteredList',result);
        });
      }
      else {
        this.set('filteredList', null);
      }
    },
    search(param) {
      if(param !== "") {
        this.store.query('rental', {city: param}).then((result) => {
          this.set('model',result);
        });
      }
      else {
        this.set('model', this.store.findAll('rental'));
      }
    }
  }
});
```

**Comment by acorncom** [comment]
@toddjordan Could you take a look at this?

**Comment by estani** [comment]
2.4 tutorial still has this issue.

`this.set('filteredList').clear();` makes no sense to me and I think it requires some explanation 
(calling a set method with no value and expecting it to return something upon which `clear()` may be called is everything but self describing, furthermore it doesn't seem to work at least in this case).

**Comment by locks** [comment]
More discussion on this https://github.com/emberjs/guides/pull/1264

**Comment by vtambourine** [comment]
Confirm the issue. It is really annoying when copypasted code from official tutorial doesn't work expected way. I suggest to add demo repository with all working examples in the very beginning of tutorial.

**Comment by toddjordan** [comment]
Putting general autocomplete improvements on my list for this week.

**Comment by toddjordan** [comment]
@ChangJoo-Park I've added a fix to the super-rentals working example repo here: https://github.com/emberjs/super-rentals/pull/12/files

I largely went with your approach but with a couple improvements: 

```javascript
import Ember from 'ember';

export default Ember.Controller.extend({
  filteredList: null,
  actions: {
    autoComplete(param) {
      if (param !== '') {
        this.get('store').query('rental', { city: param }).then((result) => this.set('filteredList', result));
      } else {
        this.set('filteredList', null);
      }
    },
    search(param) {
      if (param !== '') {
        this.store.query('rental', { city: param }).then((result) => this.set('model', result));
      } else {
        this.get('store').findAll('rental').then((result) => this.set('model', result));
      }
    }
  }
});
```

**Comment by toddjordan** [comment]
Closing as fixed by #1359
